resources/views/pages/ticket/[ticket].blade.php

```diff
@@ -0,0 +1,423 @@
+                                    @if ($ticket->Hardcore)
+                                    <button class="btn" onclick="AwardManually(1)">Award Hardcore</button>
+                                    @endif
+                                </div>
+                            @endif
+                        </div>
+                    @endif
+                </div>
+            </div>
+        </div>
+    @endif
+
+    <div class="mt-2">
+        <div class="flex w-full justify-between border-embed-highlight items-center">
+            <p role="heading" aria-level="2" class="mb-0.5 text-2xs font-bold">
+                Comments
+            </p>
+            @if ($permissions >= Permissions::Developer)
+                <div class="flex flex-col gap-y-1">
+                    <a class="btn py-2 block transition-transform lg:active:scale-[97%]" href="{!! $contactReporterUrl !!}">
+                        Contact the reporter
+                    </a>
+                </div>
+            @endif
+        </div>
+
+        <div class="commentscomponent">
+            <table id="feed" class="table-highlight"><tbody>
+                <x-comment.item
+                    :author="$ticket->reporter"
+                    :when="$ticket->ReportedAt"
+                    :payload="$ticketNotes"
+                />
+                @php $numArticleComments = getRecentArticleComments(ArticleType::AchievementTicket, $ticket->ID, $commentData) @endphp
+                @php $allowDelete = $permissions >= Permissions::Moderator @endphp
+                @foreach ($commentData as $comment)
+                    @php
+                        $when = Carbon::createFromTimestamp($comment['Submitted']);
+                        $commentUser = 
+                            ($comment['User'] === $ticket->reporter->User) ? $ticket->reporter :
+                            (($comment['User'] === $user->User) ? $user :
+                                User::firstWhere('User', $comment['User']));
+                    @endphp
+                    <x-comment.item
+                        :author="$commentUser"
+                        :when="$when"
+                        :payload="$comment['CommentPayload']"
+                        articleType="{{ ArticleType::AchievementTicket }}"
+                        :articleId="$ticket->ID"
+                        :commentId="$comment['ID']"
+                        :allowDelete="$allowDelete"
+                    />
+                @endforeach
+
+                @if (isset($user) && !$user->isMuted)
+                    {!! RenderCommentInputRow($user->User, ArticleType::AchievementTicket, $ticket->ID) !!}
+                @endif
+            </tbody></table>
+        </div>
+    </div>
+
+    <div>
+        <form method="post" action="/request/ticket/update.php">
+            {!! csrf_field() !!}
+            <input type="hidden" name="ticket" value="{{ $ticket->ID }}">
+            @if ($permissions >= Permissions::Developer)
```

Comment:
This seems like it should be `@can('manage, App\Models\User::class)`. We probably need to update `UserPolicy::requireAdministrativePrivileges()` to use a `Permissions` fallback similar to our other policies, eg:

```php
// !! untested pseudocode !!

private function requireAdministrativePrivileges(User $user, ?User $model = null): bool
{
    if (!$model) {
        return $user->hasAnyRole([
            Role::ROOT,
            Role::ADMINISTRATOR,
            Role::MODERATOR,
        ])
            || $user->getAttribute('Permissions') >= Permissions::Moderator;
    }

    // ...
}
```

Reply:
This is just preventing a user from changing the ticket state. Junior Developers have to have their solutions reviewed, so this should be restricted to Developers+. I've added an `updateState` method to `TicketPolicy`.